File: internal/schema/item.go
package schema

import (
	"astragalaxy/internal/model"

	"github.com/google/uuid"
)

type Item struct {
	ID          uuid.UUID `json:"id"`
	InventoryID uuid.UUID `json:"inventory_id"`
	Code        string    `json:"code"`
	Durability  int       `json:"durability"`
}

func ItemSchemaFromItem(item *model.Item) *Item {
	schema := Item{
		ID:          item.ID,
		InventoryID: item.InventoryID,
		Code:        item.Code,
		Durability:  item.Durability,
	}
	return &schema
}

type ItemDataResponse struct {
	Data map[string]string `json:"data"`
}

type ItemUsageResponse struct {
	Ok      bool           `json:"ok"`
	Message string         `json:"message"`
	Data    map[string]any `json:"data"`
}

type CreateItem struct {
	InventoryID uuid.UUID         `json:"inventory_id"`
	Code        string            `json:"code"`
	DataTags    map[string]string `json:"data_tags"`
}

File: astragalaxy/database/models/item.py
from typing import Any
from uuid import UUID, uuid4

from sqlalchemy import ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship

from astragalaxy.database import Base
from astragalaxy.database.models.inventory import Inventory


class Item(Base):
    __tablename__ = "items"
    id: Mapped[UUID] = mapped_column(primary_key=True, default=uuid4)
    code: Mapped[str]
    inventory_id: Mapped[UUID] = mapped_column(ForeignKey("inventories.id"))
    inventory: Mapped[Inventory] = relationship()
    data: Mapped[dict[str, Any]]
    durability: Mapped[int] = mapped_column(nullable=True, default=100)
